src/components/Form/index.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class Form extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {};
+	}
+
+	submitWrapper() {
+		if (typeof(this.props.onSubmit) !== "undefined") {
+			this.props.onSubmit();
+		}
+		alert('Form submitted successfully!');
+	}
+
+	render() {
+		return (
+			<form onSubmit={() => this.submitWrapper()}>
```

Comment:
shouldnt ```onSubmit={this.submitWrapper}``` work here?

Reply:
I see now.  Thanks for the insight sir.  I think it might be better practice to bind the function before it is called for readability.

Reply:
i think this is fine, because sometimes you want to pass down an event object or other variables, and this is a pretty readable place to put it

Reply:
event objects and other variables can be passed down even when binded though.  But yeah I guess its fine especially since its the only instance